server/Main.hs

```diff
@@ -53,31 +53,36 @@ instance A.ToJSON Error
 
 server :: [P.ExternsFile] -> P.Env -> P.Environment -> Int -> IO ()
 server externs initNamesEnv initEnv port = do
+  let toCompilerErrors :: NE.NonEmpty CST.ParserError -> Either Error a
```

Comment:
Even if we always want to feed the result of this to `Left`, I think it would be preferable to have this function be `fromParserErrors :: NE.NonEmpty CST.ParserError -> Error`. That way, this function has a clearer responsibility, and it's also clearer at the call site that we're always producing a `Left`.

Reply:
I've floated it to the top level and named it as you suggest. Thanks!